README.md

```diff
@@ -10,6 +10,53 @@ official packaging.
 
 ## Usage
 
+### Documentation
+
+The contents of `default.nix` is a function from a set of common nix packages
+to a set of functions. `default.nix` is meant to be called via the callPackage
+function. The functions it produces are as follows.
+
+#### rustc
+
+`rustc` is a function from an optional `date` (defaults to latest), and
+an optional `system` (defaults to the system nix is installed on) to
+a derivation for the rust compiler. If you wish, you can provide an
+optional hash argument to specify exactly what the blob that 
+rust-nightly-nix downloads should hash to. If no hash is provided,
+rust-nightly-nix will just trust whatever it gets.
+
+For example, 
+
+```
+let funs = callPackage "/path/to/rust-nightly.nix" { };
+  
+    rustcNightly = funs.rustc {
+      date = "2016-06-09";
+      hash = "1p0xkpfk66jq0iladqfrhqk1zc1jr9n2v2lqyf7jjbrmqx2ja65i";
+    };
+```
+
+provides a hash to add some security to the build process.
+
+#### cargo
+
+`cargo` has the same type as `rustc`.
+
+#### rustcWithSysroots
+
+`rustcWithSysroots` takes rustc as an argument, plus an optional set of
+derivations to be put in scope for the build of rustc. This is particularly
+useful if you want to make `rust-std` avalable to `rustc`.
```

Comment:
I will do that. Is the goal of the little snippet in your QuiltOS link to put multiple `rust-std` packages in scope for building rustc against?